src/validation.h

```diff
@@ -520,21 +545,40 @@ class CChainState {
      */
     mutable std::atomic<bool> m_is_ibd;
 
+    //! Manages the UTXO set, which is a reflection of the contents of `m_chain`.
+    std::unique_ptr<CoinsViews> m_coins_views;
+
 public:
+    CChainState(
+        /* parameters forwarded to CoinsViews */
+        size_t cache_size_bytes,
+        bool in_memory,
+        bool should_wipe,
+        std::string leveldb_name = "chainstate"
+        ) : m_is_ibd(false)
+    {
+        m_coins_views.reset(new CoinsViews(
+            leveldb_name, cache_size_bytes, in_memory, should_wipe));
+    }
+
     /**
      * The current chain of blockheaders we consult and build on.
      *
      * @see CChain, CBlockIndex.
      */
     CChain m_chain;
+
     /**
      * The set of all CBlockIndex entries with BLOCK_VALID_TRANSACTIONS (for itself and all ancestors) and
      * as good as our current tip or better. Entries may be failed, though, and pruning nodes may be
      * missing the data for the block.
      */
     std::set<CBlockIndex*, CBlockIndexWorkComparator> setBlockIndexCandidates;
 
-    CBlockIndex *pindexBestInvalid = nullptr;
+    CCoinsViewCache* CoinsCache() { return m_coins_views->m_cacheview.get(); }
```

Comment:
In commit "refactoring: have CCoins* data managed under CChainState" (525df322de8e6803f41a56aff16bfc1ebe95f845)

I think these three functions should be returning references instead of pointers, and maybe asserting the underlying unique_ptrs are set, since no callers are ever checking to see if the pointers are null.